index.js

```diff
@@ -14,32 +14,101 @@ const serverBuilder = require('./lib/server_builder');
 const daemonize = require('./lib/daemonize');
 const usageStats = require('./lib/stats');
 
-/**
- * Parse args
- */
+
+const express = require('express');
+const bodyParser = require('body-parser');
+const app = express();
+const url = require('url');
+
+
+var cors = require('cors');
+
+
 program.parse(process.argv);
-if (program.args.length === 0) {
-  console.error('Arguments needed, use --help');
-  process.exit();
-}
 
 /**
  * Init usage statistics
  */
-const stats = usageStats(!program.disableUsageStats, program);
-stats.track('runtime', 'init');
-stats.time('runtime', 'runtime');
+ const stats = usageStats(!program.disableUsageStats, program);
+ stats.track('runtime', 'init');
+ stats.time('runtime', 'runtime');
 
-/**
- * Validate params
- */
 const doAuthorization = !!(program.user && program.password);
 const doSecure = !!(program.key && program.certificate);
 const sessionSecret = String(+new Date()) + Math.random();
-const files = program.args.join(' ');
-const filesNamespace = crypto.createHash('md5').update(files).digest('hex');
+var files = program.args.join(' ');
+var filesNamespace = crypto.createHash('md5').update(files).digest('hex');
 const urlPath = program.urlPath.replace(/\/$/, ''); // remove trailing slash
 
+const appBuilder = connectBuilder(urlPath);
+
+const builder = serverBuilder();
+
+const io = new SocketIO({ path: `${urlPath}/socket.io` });
+
+var tailerObj = {}
+
+io.on('connection', (socket) => {
+          var ns = url.parse(socket.handshake.url, true).query.ns;
+          if(!(ns in tailerObj)){
+            tailerObj[ns] = tail([ns], {
+              buffer: program.number,
```

Comment:
What is the value of the program buffer. Can this be picked from the URL arguments? If yes we would like to have a parameter `buffer` in url args which will get used here.